### PR TITLE
Integrate veritysetup format command for dm-verity hash generation

### PR DESCRIPTION
**Pull request by martinmcconnell**
Also, include dynamic kernel version dm-verity hash in initramfs.

### CODE REVIEW

File: bin/initoverlayfs-install [hunk 1]
```diff
@@ -37,6 +37,9 @@ exec_erofs() {
     popd
     rm -f "${INITRAMFS_DIR}/initoverlayfs-$kver.img"
     mkfs.erofs $erofs_compression "${INITRAMFS_DIR}/initoverlayfs-$kver.img" ${INITRAMFS_DUMP_DIR}
+    if false; then
```
Comment (dougsland) [inline bin/initoverlayfs-install:40]: Hey Martin, I believe the if here is not needed as we have the `set -ex` in the top of the script but I am not sure, could you please double check?

**Comment by dougsland** [comment]
Hey Martin, nice patch, just small comment. You also might want to squash the patch but it's up to you.
Also, would be nice to use a title related instead of `Vroom 16770` and a short description. After that and tested, move the patch to "ready to review".

File: bin/initoverlayfs-install (hunk 1 above)
Comment (ericcurtin) [reply bin/initoverlayfs-install:40]: We should leave this in for now, it's gonna eventually become an "if ostree" kind of statement.

We don't want to do dm-verity for regular images, it has a performance cost and we don't have a secure boot story for non-ostree images.

Comment (dougsland) [reply bin/initoverlayfs-install:40]: ahh okay, thanks for the clarification.

```
OSTREE_BIN="/usr/bin/ostree"

if [ -e "$OSTREE_BIN" ]; then
       foobar
fi
```

Comment (ericcurtin) [reply bin/initoverlayfs-install:40]: We can leave if false in for now just to make progress and not block, I have ideas for the if OSTree check once the rpm-ostree integration is a bit futher along and we can boot an unverified OSTree image with initoverlayfs

Comment (ericcurtin) [reply bin/initoverlayfs-install:40]: This is one of many changes we will need in initoverlayfs-install to accommodate for the rpm-ostree flow...

**Comment by ericcurtin** [comment]
This is probably good to merge if you take it out of draft @martinmcconnell 

Just make sure the script works with both "if false" and "if true"

Next step after this PR is to make/teach systemd mount a dm-veritied initoverlayfs during bootup (and we will have to teach our special optimized Automotive init system that's coming too, but systemd is easier to start with).

**Pull request merged**